lib/new_relic/agent/instrumentation/grpc/server/instrumentation.rb

```diff
@@ -38,20 +39,26 @@ def add_http2_port_with_tracing(*args)
 
           def run_with_tracing(*args)
             set_host_and_port_and_method_info_on_desc
-
             yield
           end
 
           private
 
-          def set_host_and_port_on_server_instace(host_string)
-            info = host_string.split(':')
+          def host_and_port_from_host_string(host_string)
+            return unless host_string
+
+            info = host_string.split(':').freeze
+            return unless info.size == 2
+
+            info
+          end
 
-            # TODO:
-            # raise if info.size != 2
+          def set_host_and_port_on_server_instace(host_string)
```

Comment:
There's a slight misspelling: 
```suggestion
          def set_host_and_port_on_server_instance(host_string)
```

Reply:
The call on line 36 needs to be updated too.

Reply:
Suggested change committed, thank you! 36 updated with [7669e0b](https://github.com/newrelic/newrelic-ruby-agent/pull/1283/commits/7669e0b43f028d5f6f2f03fd5bd44bc2459eda30)